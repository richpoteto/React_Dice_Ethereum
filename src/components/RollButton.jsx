import React from 'react';
import PropTypes from 'prop-types';

function Button({ onClick, text }) {
  return (
    <button type="button" className="btn btn-primary btn-lg" onClick={onClick}>
      {text}
    </button>
  );
}
Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
};

function RollButton({ onClick }) {
  return <Button text="Roll" onClick={onClick} />;
}
RollButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default RollButton;
