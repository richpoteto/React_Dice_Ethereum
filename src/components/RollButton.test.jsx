import React from 'react';
import RollButton from './RollButton';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<RollButton onClick={(function(){})} text="text" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
