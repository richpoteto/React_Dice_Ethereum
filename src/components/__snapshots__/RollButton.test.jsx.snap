// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="btn btn-primary btn-lg"
  onClick={[Function]}
  type="button"
>
  Roll
</button>
`;
